libp2p/daemon/daemonapi.nim

```diff
@@ -755,7 +755,13 @@ proc newDaemonApi*(flags: set[P2PDaemonFlags] = {},
 
   # Starting daemon process
   # echo "Starting ", cmd, " ", args.join(" ")
-  api.process = startProcess(cmd, "", args, env, {poParentStreams})
+  api.process = 
+    try:
+      startProcess(cmd, "", args, env, {poParentStreams})
+    except CatchableError as exc:
+      raise exc
+    except Exception as exc:
+      raise newException(DaemonLocalError, exc.msg)
```

Comment:
This should be mapped to a `Defect` instead. We have no idea what this `Exception` is and the safest thing to do is to allow the process to exit here.